x-pack/filebeat/input/awss3/config_test.go

```diff
@@ -400,51 +399,79 @@ func TestConfig(t *testing.T) {
 		},
 		{
 			"backup_to_bucket with non-AWS",
-			queueURL,
 			"",
 			"",
+			nonAWSS3Bucket,
 			mapstr.M{
-				"non_aws_bucket_name":           "aBucket",
+				"non_aws_bucket_name":           nonAWSS3Bucket,
 				"non_aws_backup_to_bucket_name": "bBucket",
 				"backup_to_bucket_prefix":       "backup",
 				"number_of_workers":             5,
 			},
 			"",
 			func(queueURL, s3Bucket string, nonAWSS3Bucket string) config {
-				c := makeConfig("", s3Bucket, "")
-				c.NonAWSBucketName = "aBucket"
+				c := makeConfig("", "", nonAWSS3Bucket)
+				c.NonAWSBucketName = nonAWSS3Bucket
 				c.BackupConfig.NonAWSBackupToBucketName = "bBucket"
 				c.BackupConfig.BackupToBucketPrefix = "backup"
 				c.NumberOfWorkers = 5
 				return c
 			},
 		},
 		{
-			"error with AWS backup and non-AWS source",
-			queueURL,
+			"error with non-AWS backup and AWS source",
 			"",
+			s3Bucket,
 			"",
 			mapstr.M{
+				"bucket_arn":                    s3Bucket,
 				"non_aws_backup_to_bucket_name": "bBucket",
-				"bucket_arn":                    "arn:aws:s3:::aBucket",
 				"number_of_workers":             5,
 			},
 			"backup to non-AWS bucket can only be used for non-AWS sources",
 			nil,
 		},
 		{
-			"error with non-AWS backup and AWS source",
-			queueURL,
+			"error with AWS backup and non-AWS source",
 			"",
 			"",
+			nonAWSS3Bucket,
 			mapstr.M{
-				"non_aws_bucket_name":  "aBucket",
+				"non_aws_bucket_name":  nonAWSS3Bucket,
 				"backup_to_bucket_arn": "arn:aws:s3:::bBucket",
 				"number_of_workers":    5,
 			},
 			"backup to AWS bucket can only be used for AWS sources",
 			nil,
 		},
+		{
```

Comment:
can you add two more scenario exactly like the two you added here covering the `backup_to_bucket_arn`/`bucket_arn` scenario on top of the `non_aws_backup_to_bucket_name`/`non_aws_bucket_name`?

Reply:
we still miss a case for `backup_to_bucket_arn/bucket_arn` and same prefix